test/integration/repos_test.go

```diff
@@ -118,8 +118,8 @@ func TestRepositories_EditBranches(t *testing.T) {
 		},
 		RequiredPullRequestReviews: &github.PullRequestReviewsEnforcementRequest{
 			DismissalRestrictionsRequest: &github.DismissalRestrictionsRequest{
-				Users: []string{},
-				Teams: []string{},
+				Users: &[]string{},
+				Teams: &[]string{},
```

Comment:
This integration test (`TestRepositories_EditBranches`) still fails. I figure it could be addressed in this PR, as it covers the request being fixed. Since the test is using a personal repo rather than an org, `DismissalRestrictionsRequest` can be omitted, and then GitHub stops barfing.

FWIW, I got it to pass by making that change and also adding an enforcement URL for the `want` object:

```
enforcementURL :=  "https://api.github.com/repos/"+ *repo.Owner.Login + "/" + *repo.Name + "/branches/master/protection/enforce_admins"

...

EnforceAdmins: &github.AdminEnforcement{
			URL: &enforcementURL,
			Enabled: true,
		},
```